src/aktualizr_secondary/aktualizr_secondary_test.cc

```diff
@@ -216,92 +296,104 @@ INSTANTIATE_TEST_SUITE_P(SecondaryTestMalformedMetadata, SecondaryTestNegative,
-    auto metadata =
-        UptaneRepoWrapper().addImageFile("mytarget", _secondary->getHwId().ToString(), "non-existing-serial");
+    auto metadata = UptaneRepoWrapper().addImageFile("mytarget", secondary_->hwID().ToString(), "non-existing-serial");
 
-    EXPECT_FALSE(_secondary->putMetadata(metadata));
+    EXPECT_FALSE(secondary_->putMetadata(metadata));
   }
 
   {
     // zero targets for the ECU being tested
-    auto metadata =
-        UptaneRepoWrapper().addImageFile("mytarget", "non-existig-hwid", _secondary->getSerial().ToString());
+    auto metadata = UptaneRepoWrapper().addImageFile("mytarget", "non-existig-hwid", secondary_->serial().ToString());
 
-    EXPECT_FALSE(_secondary->putMetadata(metadata));
+    EXPECT_FALSE(secondary_->putMetadata(metadata));
   }
 }
 
 TEST_F(SecondaryTest, DirectorRootVersionIncremented) {
-  _uptane_repo.refreshRoot(Uptane::RepositoryType::Director());
-  EXPECT_TRUE(_secondary->putMetadata(_uptane_repo.getCurrentMetadata()));
+  uptane_repo_.refreshRoot(Uptane::RepositoryType::Director());
+  EXPECT_TRUE(secondary_->putMetadata(uptane_repo_.getCurrentMetadata()));
 }
 
 TEST_F(SecondaryTest, ImageRootVersionIncremented) {
-  _uptane_repo.refreshRoot(Uptane::RepositoryType::Image());
-  EXPECT_TRUE(_secondary->putMetadata(_uptane_repo.getCurrentMetadata()));
+  uptane_repo_.refreshRoot(Uptane::RepositoryType::Image());
+  EXPECT_TRUE(secondary_->putMetadata(uptane_repo_.getCurrentMetadata()));
+}
+
+TEST_F(SecondaryTest, SmallerImageFileSize) {
+  EXPECT_CALL(update_agent_, receiveData)
+      .Times((target_size - inavlid_target_size_delta) / send_buffer_size +
+             ((target_size - inavlid_target_size_delta) % send_buffer_size ? 1 : 0));
+  EXPECT_CALL(update_agent_, install).Times(1);
+
+  EXPECT_TRUE(secondary_->putMetadata(uptane_repo_.getCurrentMetadata()));
+
+  EXPECT_EQ(sendImageFile(smaller_target_), data::ResultCode::Numeric::kOk);
+  EXPECT_NE(secondary_->install(), data::ResultCode::Numeric::kOk);
 }
 
-TEST_F(SecondaryTest, InvalidImageFileSize) {
-  EXPECT_CALL(_update_agent, download).Times(1);
-  EXPECT_CALL(_update_agent, install).Times(0);
+TEST_F(SecondaryTest, BiggerImageFileSize) {
+  EXPECT_CALL(update_agent_, receiveData)
+      .Times((target_size + inavlid_target_size_delta) / send_buffer_size +
+             ((target_size + inavlid_target_size_delta) % send_buffer_size ? 1 : 0));
+  EXPECT_CALL(update_agent_, install).Times(1);
 
-  EXPECT_TRUE(_secondary->putMetadata(_uptane_repo.getCurrentMetadata()));
-  auto image_data = getImageData();
-  image_data.append("\n");
-  EXPECT_FALSE(_secondary->sendFirmware(image_data));
-  EXPECT_NE(_secondary->install(_default_target), data::ResultCode::Numeric::kOk);
+  EXPECT_TRUE(secondary_->putMetadata(uptane_repo_.getCurrentMetadata()));
+
+  EXPECT_EQ(sendImageFile(bigger_target_), data::ResultCode::Numeric::kOk);
+  EXPECT_NE(secondary_->install(), data::ResultCode::Numeric::kOk);
```

Comment:
These tests are checking that the Secondary rejects an image that is smaller or larger than what's stated in the metadata, right?

Reply:
yes